### PR TITLE
Remove and forbid use of com.google.common.collect.Sets

### PR DESCRIPTION
**Pull request by jasontedor**
This commit removes and now forbids all uses of
com.google.common.collect.Sets across the codebase. This is one of many
steps in the eventual removal of Guava as a dependency.

Relates #13224

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java [hunk 1]
```diff
@@ -32,14 +31,26 @@
 import org.elasticsearch.cluster.block.ClusterBlockException;
 import org.elasticsearch.cluster.metadata.SnapshotId;
 import org.elasticsearch.cluster.node.DiscoveryNode;
-import org.elasticsearch.cluster.routing.*;
+import org.elasticsearch.cluster.routing.IllegalShardRoutingStateException;
```
Comment (bleskes) [inline core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: why all this unpacking?

Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: It's due to different IDE settings. Some people have their IDEs set to auto-collapse imports when there are more than a few from the same package, others do not. If you'd like, I can update the pull request to not have this noise?

File: core/src/main/java/org/elasticsearch/common/util/set/Sets.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class Sets {
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/common/util/set/Sets.java]: it's a shame java need this...

File: core/src/test/java/org/elasticsearch/index/deletionpolicy/SnapshotIndexCommitExistsMatcher.java [hunk 3]
```diff
@@ -37,7 +35,8 @@
     @Override
     public boolean matchesSafely(SnapshotIndexCommit snapshotIndexCommit) {
         try {
-            HashSet<String> files = Sets.newHashSet(snapshotIndexCommit.getDirectory().listAll());
+            HashSet<String> files = new HashSet<>();
+            files.addAll(Arrays.asList(snapshotIndexCommit.getDirectory().listAll()));
```
Comment (bleskes) [inline core/src/test/java/org/elasticsearch/index/deletionpolicy/SnapshotIndexCommitExistsMatcher.java]: Sets.newHashSet(...listAll()) ?

**Comment by bleskes** [comment]
LGTM.

File: core/src/test/java/org/elasticsearch/index/deletionpolicy/SnapshotIndexCommitExistsMatcher.java (hunk 3 above)
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/index/deletionpolicy/SnapshotIndexCommitExistsMatcher.java]: Thanks. Done in 14f75571bfdc6e0e3c1efe32a613ac900b20e424.

**Comment by jasontedor** [comment]
Thanks for reviewing @bleskes.

**Pull request merged**